### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by ndikumanaisaie**
## In this milestone I covered the following:

- I first chose an API on which to base the development of the web app.
### Interactions

- Home page
   - When the page loads, the web app shows a select options field to select the continent.
   - Along with the continent name, Display a list of the Number of covid19 Active cases.
   - When the user clicks (or taps) on a country item, the application navigates to the country details page.
- Details page
   - On the details page, the web app retrieves data from the API to show detailed data about the country data.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
### Testing requirements

- Created unit tests for pure functions.
- Created integration tests for the application using the React Testing Library.

### Technical requirements

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API is stored in the Redux store.
- I filtered the data that I retrieved from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) have a unique route within the SPA.
- The project is deployed and accessible online (Link to the README file).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,15 +19,19 @@ covid19-tracker is a comprehensive up-to-date COVID-19 tracker and Treatment tra
...
 
 To get a local copy up and running follow these simple example steps.
 
 ## Setup
-clone: run `git clone https://github.com/ndikumanaisaie/ReactBookStore.git`
-Move to cloned folder: run git cd ReactBookStore.
+clone: run `git clone https://github.com/ndikumanaisaie/covid19-tracker.git`
+Move to cloned folder: run git cd covid19-tracker.
 
 ## Install
```
Comment (omar-muhamad) [inline README.md:37]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.), Please give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) at the `Acknowledgments` section in your readme file.

File: src/assets/styles/styles.css [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+html * {
+  font-family: 'Roboto', sans-serif;
+}
+
+body {
+  margin: 0;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  background-color: rgb(30, 80, 173);
+}
```
Comment (omar-muhamad) [inline src/assets/styles/styles.css]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Typography%3A%20font%20face%2C%20size%20and%20weight.), Please use the correct font families (gill sans) for body and title fonts and (lato) as control font.

image

File: src/__test__/componentTests/App.test.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      .create(
+        <MemoryRouter>
+          <Provider store={store}>
+            <App />
+          </Provider>
+        </MemoryRouter>,
+      )
+      .toJSON();
+    expect(app).toMatchSnapshot();
+  });
```
Comment (omar-muhamad) [inline src/__test__/componentTests/App.test.jsx:19]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Create%20unit%20tests%20for%20pure%20functions%20in%20your%20Redux%20code.), Please add unit tests for pure functions in your Redux code, not only snapshot tests as you did.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ndikumanaisaie,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Reem-lab — COMMENTED** [review]
# Hi @ndikumanaisaie, 👋🏻👋🏻👋🏻 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linters errors so kindly make sure to solve the linters then you can submit your PR again 🔥🚀✅

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by DammyShittu — APPROVED** [review]
Hi @ndikumanaisaie,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**